src/main/java/com/teragrep/rlo_06/RFC5424Parser.java

```diff
@@ -47,426 +47,55 @@
 
 import java.io.IOException;
 import java.io.InputStream;
-import java.nio.ByteBuffer;
+import java.util.function.Consumer;
 
 public final class RFC5424Parser {
-    private InputStream inputStream;
-    private final boolean lineFeedTermination;
-    private Boolean EOF = false;
-    private final byte[] buffer = new byte[256 * 1024];
-    private int pointer = -1;
-    private int read = -1;
+    private Stream stream;
+    private final ParserResultSet resultset;
+    private final Consumer<Stream> streamConsumer;
 
-    public RFC5424Parser(InputStream inputStream) {
-        this.inputStream = inputStream;
-        this.lineFeedTermination = true;
+    public RFC5424Parser(InputStream inputStream, RFC5424ParserSubscription subscription, RFC5424ParserSDSubscription sdSubscription) {
+        this(inputStream, subscription, sdSubscription, true);
     }
 
-    public RFC5424Parser(InputStream inputStream, boolean lineFeedTermination) {
-        this.inputStream = inputStream;
-        this.lineFeedTermination = lineFeedTermination;
+    public RFC5424Parser(InputStream inputStream, RFC5424ParserSubscription subscription, RFC5424ParserSDSubscription sdSubscription, boolean lineFeedTermination) {
+        this.stream = new Stream(inputStream);
+        this.resultset = new ParserResultSet(subscription, sdSubscription);
+
+        Priority priority = new Priority(this.resultset.PRIORITY);
+        Version version = new Version(this.resultset.VERSION);
+        Timestamp timestamp = new Timestamp(this.resultset.TIMESTAMP);
+        Hostname hostname = new Hostname(this.resultset.HOSTNAME);
+        AppName appName = new AppName(this.resultset.APPNAME);
+        ProcId procId = new ProcId(this.resultset.PROCID);
+        MsgId msgId = new MsgId(this.resultset.MSGID);
+        StructuredData structuredData = new StructuredData(this.resultset);
+        Msg msg = new Msg(this.resultset.MSG, lineFeedTermination);
+
+        this.streamConsumer = priority
```

Comment:
Any way of doing this in `priority.and(version).and(timestamp).and(hostname)...` style instead of nested rollercoaster calls?